### PR TITLE
refactor: update vc-api comply vcdm v2

### PR DESCRIPTION
**Pull request by namhoang1604**
## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [x] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description

This PR refactors the VC-api to comply the VC-api test suite

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 [vc-kit doc site](https://uncefact.github.io/vckit/)
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: packages/vc-api/src/controllers/issuer-controller.ts [hunk 1]
```diff
@@ -16,14 +16,14 @@ export const issueCredential = async (req: RequestWithAgent, res: Response) => {
     const payload = {
       ...DEFAULT_CONFIG,
       credential: req.body.credential,
-      proofFormat: req.body.options.proofFormat ?? DEFAULT_CONFIG.proofFormat,
+      proofFormat: req.body?.options?.proofFormat ?? DEFAULT_CONFIG.proofFormat,
```
Comment (huynguyen-hl) [inline packages/vc-api/src/controllers/issuer-controller.ts:19]: I noticed that the `options` field in the request payload is structured in the specifications as follows:

- mandatoryPointers [array]
- credentialId [string]

If we add an additional field like `proofFormat`, would that be correct and compliant with the specifications?

Here’s the link to the specs: https://w3c-ccg.github.io/vc-api/#issue-credential

File: packages/vc-api/src/vc-api-schemas/vc-api-v2.test-suite.yaml
Comment (huynguyen-hl) [inline packages/vc-api/src/vc-api-schemas/vc-api-v2.test-suite.yaml:1]: Could you explain a bit about this file and why it is here?
Also, could you provide a brief description of what this file does?

File: packages/vc-api/src/v2-vc-api/v2-vc-router.ts [hunk 2]
```diff
@@ -8,7 +8,10 @@ export function V2VcRouter(options: any): Router {
   const router = Router();
   router.use(VCApiDocsRouter());
   router.use(VCApiSchemaRouter(options));
-  router.use(apiKeyAuth(options.apiKey));
+  if (options.apiKey) {
+    console.log('Using API key auth');
+    router.use(apiKeyAuth(options.apiKey));
+  }
```
Comment (huynguyen-hl) [inline packages/vc-api/src/v2-vc-api/v2-vc-router.ts:14]: Could you double-check this block of code? I noticed that the `apiKeyAuth` function only accepts a single argument, which should be an object containing an `apiKey` field (`{ apiKey: string }`). So I'm not sure if passing the `apiKey` string directly to the function is correct.

image

Comment (namhoang1604) [reply packages/vc-api/src/v2-vc-api/v2-vc-router.ts:14]: Thanks @huynguyen-hl, my mistake

File: packages/vc-api/src/controllers/issuer-controller.ts (hunk 1 above)
Comment (namhoang1604) [reply packages/vc-api/src/controllers/issuer-controller.ts:19]: The options field is used to add the information when making the request, it should be fine and comply with the spec if we don't use the defined field for another purpose.

File: packages/vc-api/src/vc-api-schemas/vc-api-v2.test-suite.yaml
Comment (namhoang1604) [reply packages/vc-api/src/vc-api-schemas/vc-api-v2.test-suite.yaml:1]: Currently, we're not supporting the OAuth 2.0, so we need to make public the API to run the test suite

**Review by ldhyen99 — APPROVED** [review]

**Comment by huynguyen-hl** [comment]
@namhoang1604, Could this implementation integrate seamlessly with the credentialStatus (bitstringStatusList)?

**Comment by namhoang1604** [comment]
> @namhoang1604, Could this implementation integrate seamlessly with the credentialStatus (bitstringStatusList)?

That is quite big, I did mention to implement it in another PR

**Pull request merged**